File: src/lessons/sort/basic/bubble/AlgBubbleSort3Entity.java
package lessons.sort.basic.bubble;

import plm.universe.sort.SortingEntity;

public class AlgBubbleSort3Entity extends SortingEntity {

	public void run() {
		this.bubbleSort3();
	}

	/* BEGIN TEMPLATE */
	public void bubbleSort3() {
		/* BEGIN SOLUTION */
		for (int i = getValueCount()-1; i>=0; i--) {
			boolean swapped = false;
			for (int j = 0; j<i; j++) {
				if (!isSmaller(j,j+1)) {
					swap(j,j+1);
					swapped=true;
				}
			}
			if (!swapped) {
				return;	
			}
		}
		/* END SOLUTION */
	}
	/* END TEMPLATE */

}

File: resources/exercises/sort/basic/bubble/AlgBubbleSort3Entity.c
//RemoteSort

/* BEGIN TEMPLATE */
void bubbleSort3()  {
	/* BEGIN SOLUTION */
	int i,j;
	for (i = getValueCount()-1; i>=0; i--) {
		int swapped = 0;
		for (j = 0; j<i; j++) {
			if (!isSmaller(j,j+1)) {
				swap(j,j+1);
				swapped=1;
			}
		}
		if (!swapped) {
			return;
		}
	}
	/* END SOLUTION */
}
/* END TEMPLATE */

void run(){
	bubbleSort3();
}
